sssd_test_framework/utils/socket.py

```diff
@@ -0,0 +1,245 @@
+        """
+        Trigger the responder by running a command that activates the socket.
+
+        :param command: Command to trigger the responder (e.g., 'getent', 'sssctl').
+        :type command: str
+        :param args: Arguments for the command.
+        :type args: list[str]
+        :return: Process result of the command.
+        :rtype: ProcessResult
+        """
+        result = self.host.conn.exec([command, *args], raise_on_error=False)
+        if result.rc != 0 and "Connection refused" in result.stderr:
+            raise RuntimeError(f"Socket activation failed for {self.responder}: {result.stderr}")
+        return result
+
+
+class SSSDSocketUtils(MultihostUtility[MultihostHost]):
+    """
+    Utility class to manage SSSD socket-activated responders in tests.
+    """
+
+    def __init__(self, host: MultihostHost) -> None:
+        """
+        Initialize the socket utilities for SSSD responders.
+
+        :param host: Remote host.
+        :type host: MultihostHost
+        """
+        super().__init__(host)
+        self.responders: dict[str, SSSDSocketResponder] = {}
+
+    def get_responder(self, responder: str) -> SSSDSocketResponder:
+        """
+        Get or create an SSSDSocketResponder object for the given responder.
+
+        :param responder: Responder type (e.g., 'nss', 'pam').
+        :type responder: str
+        :return: Responder object.
+        :rtype: SSSDSocketResponder
+        """
+        if responder not in self.responders:
+            self.responders[responder] = SSSDSocketResponder(self.host, responder)
+        return self.responders[responder]
+
+    def remove_services_from_sssd_conf(self) -> None:
+        """
+        Remove the 'services' line from sssd.conf to test socket activation.
+        """
+        result = self.host.conn.exec(
+            ["sed", "-i", "/^services[[:space:]]*=.*$/d", "/etc/sssd/sssd.conf"], raise_on_error=False
+        )
+        if result.rc != 0:
+            raise RuntimeError(f"Failed to remove services line from sssd.conf: {result.stderr}")
+
+        # Restart SSSD to apply changes
+        self.host.conn.exec(["systemctl", "restart", "sssd.service"])
+
+    def restore_services_to_sssd_conf(self, services: str = "nss,pam,sudo,ssh,pac,autofs") -> None:
+        """
+        Restore the 'services' line in sssd.conf.
+
+        :param services: Comma-separated list of services to restore.
+        :type services: str
+        """
+        result = self.host.conn.exec(
+            ["sed", "-i", f"/\\[sssd\\]/a services = {services}", "/etc/sssd/sssd.conf"], raise_on_error=False
+        )
+        if result.rc != 0:
+            raise RuntimeError(f"Failed to restore services line to sssd.conf: {result.stderr}")
+
+        # Restart SSSD to apply changes
+        self.host.conn.exec(["systemctl", "restart", "sssd.service"])
+
+    def verify_socket_activation(self, responder: str, test_command: str, test_args: list[str] = []) -> bool:
```

Comment:
This looks very much like a full blown test (not something that is used to support arbitrary test), maybe it would be better to place it into a test instead?